### DISCUSSION TITLE
getserversideprops loading different slug

### DISCUSSION THREAD

**Discussion by todaysprint** [vercel/next.js#40175] [Help]
new to nextjs my url  is `https://www.todaysprint.com/questions/how-many-persons-were-born-between-x-and-v---` 
slug is "`how-many-persons-were-born-between-x-and-v--- `"
when access the page Sometimes Showing Different slug data 
example :`https://stackoverflow.com/questions/73568926/nextjs-getserversideprops-query-params-get-issue-not-properly-fetching-data`

my code is 

```
export async function getServerSideProps(ctx) {
    const  gaja  = ctx.params.gaja
    await connect();
  
    const newpaper = await blogs.find({
        $and: [
    { slug: gaja,
    type:"questions",
    mcat:"prep"} ]});
     return { props:  {user: newpaper},} 
  }
```
its happening randomly 
one page shows different page data

**Comment by todaysprint** [comment]
example 
![image](https://user-images.githubusercontent.com/110666562/188065771-053ac144-1119-4ab1-8683-151126c98feb.png)

**Comment by icyJoseph** [comment]
Hi,

Interesting, I've visited:

- https://www.todaysprint.com/questions/if-a-boat-takes-double-time-to-cover-a-distance-in-upstream-as-compare-to-t
- https://www.todaysprint.com/questions/how-many-persons-were-born-between-x-and-v---

And then, when I load both at the same time, one of them takes over... 

I notice the site is on dev mode, but that shouldn't be a problem.

I wonder if this is a case of shared JS context between requests. Since there's one server, having globals is usually a strategy prone to error because of race conditions and multiple consumers. Refer to challenge 2, https://youtu.be/sNpX8LLUgHo?t=761, on this video.

Would it be possible to take a look at the repository? You can invite me as contributor, and I could take a look. 

As a wrap up, if calls to `GSSP` could be fully scoped, as in, no closures, no globals are used, and given the same inputs, the same outputs are expected, then this shouldn't happen, but it looks to me like there's a fetching client of some sort whose data is mutated as it fetches data... These can either be easy to debug or it can take a tremendous amount of effort... Mostly because in JS objects are passed by reference, I wonder if there's some object created by the first request, and then the second request mutates it's inputs before it serialises and sends its request.
